docs/encyclopedia/temporal-nexus-security.mdx

```diff
@@ -60,4 +60,18 @@ In Temporal Cloud multiple security provisions are in place to ensure it can act
 ### Payload Encryption and Data Converter {#payload-encryption-data-converter}
 
 The Data Converter works the same for a Nexus Operation as it does for other payloads sent between a Worker and Temporal Cloud.
-There is support for a single Data Converter to be used for inbound inputs and outbound outputs, which means the encryption key often used in a custom Data Converter must be the same for both the caller and the handler.
+The caller and handler Workers must have compatible Data Converters as operation inputs and results are passed between the two.
+
+If encryption keys are used to encrypt payloads, they must be available in both the caller and handler. This may be achieved by either:
+- Using the same encryption key in both the caller and handler, for example a shared symmetric key stored in your KMS.
+- Using different encryption keys and passing the KMS encryption key ID via payload metadata.
```

Comment:
Isn't this just using the same encryption keys on both sides? I would remove this bullet point until we have a better story around per service payload converters.

Reply:
If you're adding a Nexus service/handler to an existing worker, you may have an existing encryption key for your data converter. The caller workflow may have a different encryption key in their worker. If they must encrypt with the same key, then something has to change, or just walk away since it doesn't seem like a fit right now.

But you don't have to encrypt payloads with the same encryption keys on both sides, the caller and handler can keep using whatever existing keys they have, and just flow the KMS key id and if the other side has permissions to get that key, they can decrypt the payload and it just works.

We can definitely update when we get a better story around per service payload converters, but think what we have right now will work for many folks, esp. intra org -- so want to make sure folks are not dissuaded from using Nexus due to a perceived limitation.

Reply:
Yes, I just think that this comment is more confusing than helpful.
I would keep it at for now:

```
Using the same encryption key in both the caller and handler, for example a shared symmetric key stored in your KMS.
```